lib/modules/platform/codecommit/codecommit-client.ts

```diff
@@ -306,37 +293,47 @@ export async function createPrApprovalRule(
 }
 
 export function getCodeCommitUrl(
-  region: string,
-  repoName: string,
-  credentials: Credentials
+  repoMetadata: RepositoryMetadata,
+  repoName: string
 ): string {
-  const signer = new aws4.RequestSigner(
-    {
-      service: 'codecommit',
-      host: `git-codecommit.${region}.amazonaws.com`,
-      method: 'GIT',
-      path: `v1/repos/${repoName}`,
-    },
-    credentials
-  );
+  logger.debug('get code commit url');
+  if (!process.env.AWS_ACCESS_KEY_ID || !process.env.AWS_SECRET_ACCESS_KEY) {
+    if (repoMetadata.cloneUrlHttp) {
+      return repoMetadata.cloneUrlHttp;
+    }
+    // shouldn't reach here, but just in case
+    return `https://git-codecommit.${
+      process.env.AWS_REGION ?? 'us-east-1'
```

Comment:
So if IAM roles are in use, and `repoMetadata` does not include the URL, then we default to `us-east-1`?

Reply:
We default to AWS_REGION.

And if that doesnt exist then `us-east-1` which is aws behaviour 
But i believe the code fails before if AWS_Region is not set.

Ill check what exactly fails if there is no region this edge case needs to be checked

Reply:
listRepositories fails if AWS_REGION is not available, code will quit at InitPlatform